src/bosh-director/lib/bosh/director/addon/filter.rb

```diff
@@ -106,6 +116,15 @@ def has_applicable_network?(deployment_instance_group)
           deployment_instance_group.has_network?(network_name)
         end
       end
+
+      def has_teams?
+        !@applicable_teams.nil? && !@applicable_teams.empty?
+      end
+
+      def has_applicable_team?(deployment_teams)
```

Comment:
the `has_teams?`and `has_applicable_team?` methods look to be mirroring the stemcell implementation which introduces a bit of necessary complexity. Could we remove the extra calls to `has_teams?` that appear in `has_applicable_team?`.

Reply:
I believe each method should check the input it used to avoid problems with nil array. has_applicable_team? method uses @applicable_teams array, so I think it's better to have the check